satpy/readers/generic_image.py

```diff
@@ -117,4 +136,24 @@ def mask_image_data(data):
                                 for i in range(data.shape[0])])
         data.data = masked_data
         data = data.sel(bands=BANDS[data.bands.size - 1])
+    elif hasattr(data, 'nodatavals') and data.nodatavals:
+        try:
+            nodata_handling = info['nodata_handling'] if info else NODATA_HANDLING_FILLVALUE
+        except KeyError:
+            nodata_handling = NODATA_HANDLING_FILLVALUE
```

Comment:
Assuming `info` is never `None` as it is always passed, then this block could be simplified to:

```suggestion
        nodata_handling = info.get('nodata_handling', NODATA_HANDLING_FILLVALUE)
```

Reply:
Yes, would be much nicer. If we consider `mask_image_data` as private function and info as not none, I can change this.